python/bindings/openravepy_trajectory.cpp

```diff
@@ -353,6 +346,42 @@ object PyTrajectoryBase::SamplePointsSameDeltaTime2D(dReal deltatime,
     return this->SamplePointsSameDeltaTime2D(deltatime, ensureLastPoint, pyspec);
 }
 
+object PyTrajectoryBase::SampleRangeSameDeltaTime2D(dReal deltatime,
+                                                    dReal startTime,
+                                                    dReal stopTime,
+                                                    bool ensureLastPoint) const
+{
+    std::vector<dReal>& values = _vdataCache;
```

Comment:
@kanbouchou are you sure that `SampleRangeSameDeltaTime2D` won't be called from multiple threads? Users might think that a sample function might be safe to call from multiple threads...
perhaps can use thread local storage via `boost::thread_specific_ptr`?

Reply:
> are you sure that SampleRangeSameDeltaTime2D won't be called from multiple threads?

It depends on the user, so I'm not 100% sure. Thread safety property of `PyTrajectoryBase` class hasn't changed in this MR though, `GetWaypoints` and others are using `_vdataCache` as well.

If we want to ensure thread-safety of concurrent sampling / waypoint getting on this side, we could use thread-local stuff, or just a mutex. Or, we explicitly document that this class is not thread-safe and user would have to ensure it on the application side. What do you think?

Reply:
Hmm... problem is that we don't have a good model of how users might be using trajectories. So although it might not be the case that they sample from the same trajectory in different threads, there's no guarantee.
In this case, temporary cache is probably best handled with thread local storage...

Reply:
I'll make sampling and waypoints reading multi thread safe.

Reply:
Made caches TLS and also added comment about thread safety in class description. Please check again.